src/NuGet.Core/NuGet.Protocol/RemoteRepositories/RemoteV3FindPackageByIdResource.cs

```diff
@@ -150,22 +168,35 @@ public RemoteV3FindPackageByIdResource(SourceRepository sourceRepository, HttpSo
                 throw new ArgumentNullException(nameof(logger));
             }
 
-            cancellationToken.ThrowIfCancellationRequested();
-
-            var packageInfo = await GetPackageInfoAsync(id, version, cacheContext, logger, cancellationToken);
-            if (packageInfo == null)
+            var stopwatch = Stopwatch.StartNew();
+            try
             {
-                return null;
-            }
+                cancellationToken.ThrowIfCancellationRequested();
 
-            var reader = await _nupkgDownloader.GetNuspecReaderFromNupkgAsync(
-                packageInfo.Identity,
-                packageInfo.ContentUri,
-                cacheContext,
-                logger,
-                cancellationToken);
+                var packageInfo = await GetPackageInfoAsync(id, version, cacheContext, logger, cancellationToken);
+                if (packageInfo == null)
+                {
+                    return null;
+                }
+
+                var reader = await _nupkgDownloader.GetNuspecReaderFromNupkgAsync(
+                    packageInfo.Identity,
+                    packageInfo.ContentUri,
+                    cacheContext,
+                    logger,
+                    cancellationToken);
 
-            return GetDependencyInfo(reader);
+                return GetDependencyInfo(reader);
+            }
+            finally
```

Comment:
This will never be used, but anyways, this is where the actual download happens.